util/csvresponse.py

```diff
@@ -86,14 +88,32 @@ def _create_filter_list(self,keys):
         # start with fields we never want to include
         drop = {'id', 'annotations'}
         for key in keys:
-            # remove any "extra" keys while keeping qc params and removing 'provenance' params
-            if (key not in params_to_include and not self._is_qc_param(key)) or 'provenance' in key:
+            # remove any "extra" keys while keeping relevant qc params and removing 'provenance' params
+            if self._is_qc_param(key):
+                # only include if a requested param matches the QC Key
+                if not self._needs_qc_param(key,params_to_include):
+                    drop.add(key)
+            elif key not in params_to_include or 'provenance' in key:
                 drop.add(key)
         return drop
 
+    @staticmethod
+    def _needs_qc_param(qc_param,params_to_include):
+        """
+        Determines if the specified param key corresponds to one of the requested parameters.
+        :param qc_param: the QC
+        :param params_to_include: list containing requested parameters
+        :return: True if the QC param should be included, False otherwise
+        """
+        # the qc param ends with either 'qc_executed' or 'qc_results', but not both...
+        # find the position of the '_' preceding the 'qc'
+        position = max(qc_param.rfind(QC_EXECUTED),qc_param.rfind(QC_RESULTS)) - 1
+        assoc_param = qc_param[0:position]
```

Comment:
Did you consider:
```python
assoc_param = split(qc_param, '_qc_')[0]
```

Reply:
OK, I was thinking Java where the String.split() function uses a regex to determine the split changes. I like yours much better and will implement.